envoy/common/backoff_strategy.h

```diff
@@ -28,6 +28,14 @@ class BackOffStrategy {
    * @param base_interval the new base interval for the backoff strategy.
    */
   virtual void reset(uint64_t base_interval) PURE;
+
+  /**
+   * Checks if a time interval is greater than the maximum time interval configured for a backoff
+   * strategy.
+   * @param interval time interval to be checked.
+   * @return returns true if interval is greater than the maximum time interval
+   */
+  virtual bool isOverTimeLimit(uint64_t interval_ms) PURE;
```

Comment:
The maximum time interval doesn't seem to be a shared property among all backoff strategies, but very specific to `JitteredExponentialBackOffStrategy`. I think this PR can be redesigned without introducing this to the BackoffStrategy interface.

Reply:
done, redesigned the PR to use the derived class directly